src/main/java/jp/co/soramitsu/sora/crypto/service/impl/JSONVerifierImpl.java

```diff
@@ -0,0 +1,44 @@
+package jp.co.soramitsu.sora.crypto.service.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.security.Signature;
+import java.security.SignatureException;
+import jp.co.soramitsu.sora.common.Util;
+import jp.co.soramitsu.sora.crypto.common.Consts;
+import jp.co.soramitsu.sora.crypto.proof.Proof;
+import jp.co.soramitsu.sora.crypto.service.JSONCanonizer;
+import jp.co.soramitsu.sora.crypto.service.JSONVerifier;
+import lombok.AllArgsConstructor;
+
+
+@AllArgsConstructor
+public class JSONVerifierImpl implements JSONVerifier {
+
+  private final JSONCanonizer canonizer;
+  private final ObjectMapper mapper;
+
+  @Override
+  public boolean verify(Object object, Signature signature)
+      throws IOException, SignatureException {
+    ObjectNode root = mapper.valueToTree(object);
+
+    if (!root.hasNonNull(Consts.PROOF_KEY)) {
+      return false;
+    }
+
+    // read proof
+    JsonNode proofNode = root.get(Consts.PROOF_KEY);
```

Comment:
I am not sure if it works like this

Reply:
```
/**
     * Method for accessing value of the specified field of
     * an object node. If this node is not an object (or it
     * does not have a value for specified field name), or
     * if there is no field with such name, null is returned.
     *<p>
     * NOTE: if the property value has been explicitly set as <code>null</code>
     * (which is different from removal!),
     * a {@link com.fasterxml.jackson.databind.node.NullNode} will be returned,
     * not null.
     *
     * @return Node that represent value of the specified field,
     *   if this node is an object and has value for the specified
     *   field. Null otherwise.
     */
```

it is just safer to do hasNonNull check first